duckduckgo/macos-browser
# Modularize debug menu and login items logic

Task/Issue URL: https://app.asana.com/0/0/1205112155849082/f

## Stack:
- https://github.com/duckduckgo/macos-browser/pull/1378 <- This PR
   - https://github.com/duckduckgo/macos-browser/pull/1389
   - https://github.com/duckduckgo/macos-browser/pull/1397

## Description

This PR:
- Moves the debug logic that's currently in `NetworkProtectionTunnelProvider` into its own class.
- Moves the login item management logic into its own class.

This will have the following advantages:
- Since most of the debug logic is platform specific, moving it out of `NetworkProtectionTunnelProvider` means we're closer to being able to share it with iOS, which IMHO would be great many ways.
- I'll be able to tackle fixes to the login items quicker, without having to deal with unrelated code.  Additionally I'll be able to add unit tests more easily.

## Testing

### Test 1: Debug Menu

Test the debug menu and make sure all works well.

1. Reset NetP's status.
2. Remove the system extension and login items.
3. Change your selected server
4. Simulate a failure.

### Test 2: Login Items

1. Reset NetP's status.
2. Launch the app.
3. Start NetP.
4. Make sure the login items are launched. 

---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
